tensorflow/compiler/mlir/lite/ir/tfl_ops.cc

```diff
@@ -1443,12 +1443,66 @@ void FakeQuantOp::getCanonicalizationPatterns(OwningRewritePatternList &results,
 
 // TODO(b/133486129): Implement shape inference for unpack
 
-static LogicalResult Verify(UnpackOp op) {
-  // TODO(antiagainst): Implement other checks as in
-  // tensorflow/lite/kernels/unpack.cc
+LogicalResult UnpackOp::inferReturnTypes(
+    MLIRContext *context, Optional<Location> location, ValueRange operands,
+    DictionaryAttr attributes, RegionRange regions,
+    SmallVectorImpl<Type> &inferredReturnTypes) {
+  auto num = attributes.get("num");
```

Comment:
I should have mentioned op adaptors here (e.g., you can do UnpackOp::Adaptor op(operands, attributes); and then use its verify method and named accessors). I need to fix up another one too, so just leave "TODO(jpienaar): Use Adaptor instead" here.

Reply:
Done :-)